src/modules/solution/solution.module.ts

```diff
@@ -0,0 +1,11 @@
+import { Module } from '@nestjs/common';
+import { JwtStrategy } from '../core/auth/jwt/jwt.strategy';
+import { PrismaService } from '../core/database/prisma/prisma.service';
+import { SolutionController } from './controllers/solution.controller';
+import { SolutionService } from './services/solution.service';
+
+@Module({
+  controllers: [SolutionController],
+  providers: [SolutionService, PrismaService, JwtStrategy]
```

Comment:
JwtStrategy 없어도 되지 않나요???

Reply:
Provider에 작성해줘야 한다. 라고 알고 있는데 필요없나요?

Reply:
알아볼 기회가 있으면 공유 드리도록 하겠습니다
가드랑 패스포트쪽이 워낙 블랙박스형태라 볼때마다 헷갈리네요

Reply:
PassportModule을 import하고 있는 AuthModule의 provider로 JwtStrategy를 이미 등록해 주었고, 각각의 controller에서 UseGuards를 사용할 때 AuthGuard를 extends하고 있는 JwtAuthGuard "클래스"를 직접 넣어주기 때문에 따로 다른 모듈들의 module.ts의 provider로 등록해 주지 않아도 되는 것 아닐까요?

Reply:
AuthModule에서 등록했기때문에 더 이상 주입하지 않아도 된다는 뜻일까요?
AuthGuard가 싱글톤으로 동작한다면 가능할 것 같습니다. 당연히 Guard도 싱글톤이겠죠?(질문입니다)

Reply:
nest js guard 공식문서에 보면 

"Above, we passed the RolesGuard class (instead of an instance), leaving responsibility for instantiation to the framework and enabling dependency injection. As with pipes and exception filters, we can also pass an in-place instance" 

라고 명시되어 있는데 이를 쉽게 풀어 쓰면 "UserGuard의 인자로 넘어온 JwtAuthGuard 클래스에 대한 인스턴스화는 nestjs가 알아서 해줍니다" 인데

위 문서를 보았을 때 드는 생각은 nestjs가 UseGuard에 JwtAuthGuard가 들어오게 된다면 싱글톤으로 생성해 놓고 각 controller의 라우터마다 메타데이터를 생성해 런타임 때 요청이 들어오게 된다면 nest 내부의 메타데이터를 확인하고 가드를 적용시켜주는 것이 아닐까라는 생각이 드네요.